python.d/tomcat.chart.py

```diff
@@ -3,74 +3,145 @@
+        'lines': [
+            ["free", 'free', "absolute", 1, 1048576],
+            ["eden_used", 'eden', "absolute", 1, 1048576],
+            ["survivor_used", 'survivor', "absolute", 1, 1048576],
+            ["tenured_used", 'tenured', "absolute", 1, 1048576],
+            ["code_cache_used", 'code cache', "absolute", 1, 1048576],
+            ["compressed_used", 'compressed', "absolute", 1, 1048576],
+            ["metaspace_used", 'metaspace', "absolute", 1, 1048576],
+        ]},
+    'jvm_eden': {
+        'options': [None, "Eden Memory Usage", "MB", "memory", "tomcat.jvm_eden", "area"],
         'lines': [
-            ["free", None, "absolute", 1, 1048576]
-        ]}
+            ["eden_used", 'used', "absolute", 1, 1048576],
+            ["eden_commited", 'commited', "absolute", 1, 1048576],
+            ["eden_max", 'max', "absolute", 1, 1048576]
+        ]},
+    'jvm_survivor': {
+        'options': [None, "Survivor Memory Usage", "MB", "memory", "tomcat.jvm_survivor", "area"],
+        'lines': [
+            ["survivor_used", 'used', "absolute", 1, 1048576],
+            ["survivor_commited", 'commited', "absolute", 1, 1048576],
+            ["survivor_max", 'max', "absolute", 1, 1048576]
+        ]},
+    'jvm_tenured': {
+        'options': [None, "Tenured Memory Usage", "MB", "memory", "tomcat.jvm_tenured", "area"],
+        'lines': [
+            ["tenured_used", 'used', "absolute", 1, 1048576],
+            ["tenured_commited", 'commited', "absolute", 1, 1048576],
+            ["tenured_max", 'max', "absolute", 1, 1048576]
+        ]},
 }
 
-
 class Service(UrlService):
     def __init__(self, configuration=None, name=None):
         UrlService.__init__(self, configuration=configuration, name=name)
         self.url = self.configuration.get('url', "http://127.0.0.1:8080/manager/status?XML=true")
+        self.connector_name = self.configuration.get('connector_name', None)
         self.order = ORDER
         self.definitions = CHARTS
 
-    def check(self):
-        netloc = urlparse(self.url).netloc.rpartition(':')
-        if netloc[1] == ':': port = netloc[2]
-        else: port = 80
-        
-        self.regex_jvm = compile(r'<jvm>.*?</jvm>')
-        self.regex_connector = compile(r'[a-z-]+%s.*?/connector' % port)
-        self.regex = compile(r'([\w]+)=\\?[\'\"](\d+)\\?[\'\"]')
-        
-        return UrlService.check(self)
-
     def _get_data(self):
         """
         Format data received from http request
         :return: dict
         """
-        data = self._get_raw_data()
-        if data:
-            jvm = self.regex_jvm.findall(data) or ['']
-            connector = self.regex_connector.findall(data) or ['']
-            data = dict(self.regex.findall(''.join([jvm[0], connector[0]])))
-        
-        return data or None
+        data = None
+        raw_data = self._get_raw_data()
+        if raw_data:
+            xml = ET.fromstring(raw_data)
```

Comment:
this one will fail if data is not xml (`url` without `?XML=true`). So i suggest to wrap it in `try/except` (`ET.ParseError`) and add a debug info to warn the user we need xml (`self.debug('TEXT')`)